packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue

```diff
@@ -0,0 +1,436 @@
+		port: {
+			id: uuidv4(),
+			label: `${portLabel} Result`,
+			type: FunmanOperation.outputs[0].type,
+			value: runId
+		}
+	});
+};
+
+const addConstraintForm = () => {
+	const state = _.cloneDeep(props.node.state);
+	const newGroup: ConstraintGroup = {
+		borderColour: '#00c387',
+		name: '',
+		timepoints: { lb: 0, ub: 100 },
+		variables: []
+	};
+	if (!state.constraintGroups) {
+		state.constraintGroups = [];
+	}
+	state.constraintGroups.push(newGroup);
+
+	workflowEventBus.emitNodeStateChange({
+		workflowId: props.node.workflowId,
+		nodeId: props.node.id,
+		state
+	});
+};
+
+const deleteConstraintGroupForm = (data) => {
+	const state = _.cloneDeep(props.node.state);
+	if (!state.constraintGroups) {
+		return;
+	}
+	state.constraintGroups.splice(data.index, 1);
+
+	workflowEventBus.emitNodeStateChange({
+		workflowId: props.node.workflowId,
+		nodeId: props.node.id,
+		state
+	});
+};
+
+const updateConstraintGroupForm = (data) => {
+	const state = _.cloneDeep(props.node.state);
+	if (!state.constraintGroups) {
+		return;
+	}
+	state.constraintGroups[data.index] = data.updatedConfig;
+
+	workflowEventBus.emitNodeStateChange({
+		workflowId: props.node.workflowId,
+		nodeId: props.node.id,
+		state
+	});
+};
+
+// Used to set requestStepList.
+// Grab startTime, endTime, numberOfSteps and create list.
+function getStepList() {
+	const aList = [startTime.value];
+	const stepSize = (endTime.value - startTime.value) / numberOfSteps.value;
+	for (let i = 1; i < numberOfSteps.value; i++) {
+		aList[i] = i * stepSize;
+	}
+	aList.push(endTime.value);
+	return aList;
+}
+
+// Set model, modelConfiguration, modelNodeOptions
+watch(
+	() => props.node.inputs[0],
+	async () => {
+		const modelConfigurationId = props.node.inputs[0].value?.[0];
+		if (modelConfigurationId) {
+			modelConfiguration.value = await getModelConfigurationById(modelConfigurationId);
+			if (modelConfiguration.value) {
+				model.value = await getModel(modelConfiguration.value.modelId);
+				const modelColumnNameOptions: string[] =
+					modelConfiguration.value.configuration.model.states.map((state) => state.id);
+				// add observables
+				if (modelConfiguration.value.configuration.semantics?.ode?.observables) {
+					modelConfiguration.value.configuration.semantics.ode.observables.forEach((o) => {
+						modelColumnNameOptions.push(o.id);
+					});
+				}
+				modelNodeOptions.value = modelColumnNameOptions;
+
+				if (model.value && model.value.semantics?.ode.parameters) {
+					requestParameters.value = model.value.semantics?.ode.parameters.map((ele) => {
+						if (ele.distribution) {
+							return {
+								name: ele.id,
+								interval: {
+									lb: ele.distribution.parameters.minimum,
+									ub: ele.distribution.parameters.maximum
+								},
+								label: 'any'
+							};
+						}
+
+						return {
+							name: ele.id,
+							interval: {
+								lb: ele.value,
+								ub: ele.value
+							},
+							label: 'any'
+						};
+					});
```

Comment:
- Have we checked that Funman can even process observables?
- I would recommend that we move the parsing out into its own function, makes things easier to read and to do tests.

Reply:
good catch on observables.

Good call to throw this into helpers I have made
setModelOptions, setRequestParameters